src/UniswapV2LpTokenCallee.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2020 Maker Ecosystem Growth Holdings, INC.
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+
+interface VatLike {
+    function hope(address) external;
+}
+
+interface GemJoinLike {
+    function dec() external view returns (uint256);
+    function gem() external view returns (TokenLike);
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+    function vat() external view returns (VatLike);
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
```

Comment:
Should we name this LpTokenLike? This way it's different from the other contract.

Reply:
There are interactions with tokens that aren't LP in that contract. I'm gonna try to create a separate interface that inherits from that.